pkg/cloudprovider/cloudapi/azure/azure_compute.go

```diff
@@ -309,19 +314,25 @@ func (computeCfg *computeServiceConfig) GetInventoryStats() *internal.CloudServi
 	return computeCfg.inventoryStats
 }
 
-func (computeCfg *computeServiceConfig) ResetCachedState() {
-	computeCfg.SetResourceFilters(nil)
+func (computeCfg *computeServiceConfig) ResetInventoryCache() {
+	computeCfg.resourcesCache.UpdateSnapshot(nil)
 	computeCfg.inventoryStats.ResetInventoryPollStats()
 }
 
-func (computeCfg *computeServiceConfig) UpdateServiceConfig(newConfig internal.CloudServiceInterface) {
+func (computeCfg *computeServiceConfig) UpdateServiceConfig(newConfig internal.CloudServiceInterface) error {
 	newComputeServiceConfig := newConfig.(*computeServiceConfig)
 	computeCfg.nwIntfAPIClient = newComputeServiceConfig.nwIntfAPIClient
 	computeCfg.nsgAPIClient = newComputeServiceConfig.nsgAPIClient
 	computeCfg.asgAPIClient = newComputeServiceConfig.asgAPIClient
 	computeCfg.vnetAPIClient = newComputeServiceConfig.vnetAPIClient
 	computeCfg.resourceGraphAPIClient = newComputeServiceConfig.resourceGraphAPIClient
 	computeCfg.credentials = newComputeServiceConfig.credentials
+	for _, value := range computeCfg.selectors {
```

Comment:
And a new function for updating existing selector.. Also, add a comment, when we need to update the selectors, like if credentials have changed to new subscription or region update has happened. In both cases, the stored filters of queries, needs to be resetted

Reply:
I think we should keep it as is due to the following reasons.
- This function is meant for updating service config and filters are a part of service config.
- If I add another function in CloudServiceInterface interface for updating filters, for AWS its a no-op, there update function is never required.